src/rust/engine/process_execution/src/lib.rs

```diff
@@ -93,62 +92,63 @@ impl Platform {
   }
 }
 
-#[derive(PartialOrd, Ord, Clone, Copy, Debug, Eq, PartialEq, Hash)]
-pub enum PlatformConstraint {
-  Darwin,
-  Linux,
-  None,
-}
-
-impl PlatformConstraint {
-  pub fn current_platform_constraint() -> Result<PlatformConstraint, String> {
-    Platform::current().map(|p: Platform| p.into())
-  }
-}
-
-impl From<Platform> for PlatformConstraint {
-  fn from(platform: Platform) -> PlatformConstraint {
+impl From<Platform> for String {
+  fn from(platform: Platform) -> String {
     match platform {
-      Platform::Linux => PlatformConstraint::Linux,
-      Platform::Darwin => PlatformConstraint::Darwin,
+      Platform::Linux => "linux".to_string(),
+      Platform::Darwin => "darwin".to_string(),
     }
   }
 }
 
-impl TryFrom<&String> for PlatformConstraint {
+impl TryFrom<&String> for Platform {
   type Error = String;
-  ///
-  /// This is a helper method to convert values from the python/engine/platform.py::PlatformConstraint enum,
-  /// which have been serialized, into the rust PlatformConstraint enum.
-  ///
   fn try_from(variant_candidate: &String) -> Result<Self, Self::Error> {
     match variant_candidate.as_ref() {
-      "darwin" => Ok(PlatformConstraint::Darwin),
-      "linux" => Ok(PlatformConstraint::Linux),
-      "none" => Ok(PlatformConstraint::None),
+      "darwin" => Ok(Platform::Darwin),
+      "linux" => Ok(Platform::Linux),
       other => Err(format!(
-        "Unknown, platform {:?} encountered in parsing",
+        "Unknown platform {:?} encountered in parsing",
         other
       )),
     }
   }
 }
 
-impl From<Platform> for String {
-  fn from(platform: Platform) -> String {
-    match platform {
-      Platform::Linux => "linux".to_string(),
-      Platform::Darwin => "darwin".to_string(),
-    }
+#[derive(PartialOrd, Ord, Clone, Copy, Debug, Eq, PartialEq, Hash)]
+pub struct PlatformConstraint {
+  platform: Option<Platform>,
```

Comment:
This was baked into PlatformConstraint previously, but it does seem odd to have a single field with optionality baked in here. At use sites this reads "I have a required field ... but that (wrapped) field is optional." It would seem the same could be achieved with no loss of clarity by having the use site just say its an Option<Platform> doing away with this type altogether. Maybe more knobs were planned for a `PlatformConstraint` but never materialized?

Reply:
Very true. I'd be happy to remove `PlatformConstraint` and make it `Optional[Platform]` at call sites.

It looks like we're not using `PlatformConstraint` in any `@rules`, so no need to "newtype" it.

Any context we're missing @stuhood @hrfuller?

Reply:
I ended up putting the context on the main PR thread rather than here... so yea, can refer to that. IIRC, `Platform` was a concrete single platform, while `PlatformConstraint` was meant to select multiple `Platform`s.

I think that pushing further the [PEP425](https://www.python.org/dev/peps/pep-0425/) parallel while working on https://docs.google.com/document/d/1fFjMlQ7fLKsq3bomaoDoNgMxL2zMZPiqk0wjqZX8AFM/edit#heading=h.pquqi3lvc4go might be useful... @jsirois would know better, but my understanding of the PEP425 tags is that when they are partially specified they represent a set of platforms that share the specified portion of the tag. So not a range, per-se... more like a prefix match maybe?

Reply:
@stuhood that's how things work in spirit, but you cannot actually partially specify a tag, it must be a complete tag. That tag can contain elements that make it looser though like `any` and `none` in certain slots. For the python ecosystem, the flecibility is provided by platform sets. The match within a set must be exact, but the set itself contains both exact platform matches and looser ones too. Unfortunately the logic for generating the sets is not codified in any PEP I know and the defacto standard only exists in https://github.com/pypa/packaging/blob/master/packaging/tags.py#L805.

Reply:
The idea was to have a concrete type to wrap a value of None to indicate that no platform constraint was applied. The ergonomics of that were supposed to be that None could be confusing by itself as the platform constraint, but in retrospect the class doesn't offer much beside a nicer name around None. I'm fine with the simplification as the intended purpose doesn't seem to have been clear anyway.